website/index.html

```diff
@@ -60,36 +38,62 @@ <h2>Echo</h2>
 <script type="text/javascript">if (!mobileConsole.status.initialized) {mobileConsole.init();}</script>
 -->
 
-<script type="text/javascript">console.log("Loading brython...")</script>
-<script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.9.1/brython.min.js"></script>
+<!-- It is important to remember that all text/python scripts are loaded and run in order AFTER all the javascript
+     scripts. You cannot interleave python/javascript scripts and expect them to run in order.
+-->
 
+<!-- Load the javascript loader functions -->
+<script src="loader.js"></script>
 <script type="text/javascript">
-  console.log("Loading brython_stdlib...");
-  document.getElementById("psychology").innerHTML = "Loading Standard Libraries...";
+  start_loader("Loading System...")
 </script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.9.1/brython.min.js"></script>
+
+<!-- If you know your users will be on a slow system and will notice the download of brython and the std library
+     the first time then it may be worth showing an updated loader message. On subsequent runs these are likely to
+     be cached and will load instantly, leading to this message flashing too fast for the user to read.
+
+     Generally it may be better to leave a single "Loading System..." message and not try to update it at all.
+-->
+<!--<script type="text/javascript">-->
+<!--  update_loader_message("Loading Standard Library...")-->
+<!--</script>-->
 <script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.9.1/brython_stdlib.min.js"></script>
 <!-- Use the following two to replace the above CDN, if you want to use not-yet-released Brython code base -->
-<!--
-<script src="https://raw.githack.com/brython-dev/brython/master/www/src/brython.js"></script>
-<script src="https://raw.githack.com/brython-dev/brython/master/www/src/brython_stdlib.js"></script>
--->
-    <script type="text/python">
-      print("Loading main python program...")
-      from browser import document
-      document["psychology"].text = "Loading Main Program..."
-    </script>
+<!--<script src="https://raw.githack.com/brython-dev/brython/master/www/src/brython.js"></script>-->
+<!--<script src="https://raw.githack.com/brython-dev/brython/master/www/src/brython_stdlib.js"></script>-->
+
+<!-- On the first run, the parsing of your main.py code below could take a few seconds. The user will
+     see whatever loader message is on the page when this process begins. On subsequent runs this code will
+     have been cached and the modules it uses will already have been transpiled. It will go fast and any message
+     you change will flash quickly.
+ -->
+
+<!--    <script type="text/javascript">-->
+<!--      update_loader_message("Loading Program Code....")-->
+<!--    </script>-->
```

Comment:
Again, on my system, after I enabling this message, I'll see the previous 2 messages being quickly replaced by this message, which will stay for a couple seconds. Yet I would still suggest to enable it as the default.
```suggestion
    <script type="text/javascript">
      update_loader_message("Loading Program Code....")
    </script>
```

Reply:
Will do!